presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java

```diff
@@ -644,4 +648,15 @@ public static void dropTableIfExists(QueryRunner queryRunner, String catalogName
     {
         queryRunner.execute(format("DROP TABLE IF EXISTS %s.%s.%s", catalogName, schemaName, tableName));
     }
+
+    protected String normalizeIdentifier(String name, String catalogName)
+    {
+        Metadata metadata = getQueryRunner().getMetadata();
+        TransactionId txid = getQueryRunner().getTransactionManager().beginTransaction(false);
+        Session session = getSession().beginTransactionId(txid, getQueryRunner().getTransactionManager(), new AllowAllAccessControl());
+        CatalogManager catalogManager = getDistributedQueryRunner().getCoordinator().getCatalogManager();
+        ConnectorId connectorId = catalogManager.getCatalog(catalogName).get().getConnectorId();
+
+        return metadata.normalizeIdentifier(session, connectorId.getCatalogName(), name);
```

Comment:
```suggestion
        String result = metadata.normalizeIdentifier(session, catalogName, name);
        getQueryRunner().getTransactionManager().asyncAbort(txid);
        return result;
```

As I understand, `catalog.getConnectorId().getCatalogName()` is always the same as `catalogName`, so do you think it makes sense to use `catalogName` here directly? 

Besides, is it better to abort the transaction explicitly? Since the transaction manager in query runner is a long lived instance through the whole test class scope, and I observed nearly one hundred additional un-closed transactions after this change when running `TestIcebergDistributedHive`.

Reply:
Thanks for pointing it out. I have made the changes.